soapui/src/main/java/com/eviware/soapui/impl/rest/actions/explorer/callback/EndpointExplorerCallback.java

```diff
@@ -0,0 +1,286 @@
+        Analytics.trackAction(EXPLORE_API_CHANGE_HTTP_METHOD, "ChangeMethodTo", changeMethodTo);
+    }
+
+    public void exploreAPIClickAuthHeadersTab() {
+        Analytics.trackAction(EXPLORE_API_CLICK_AUTH_HEADERS_TAB);
+    }
+
+    public void exploreAPIClickBodyTab() {
+        Analytics.trackAction(EXPLORE_API_CLICK_BODY_TAB);
+    }
+
+    private String extractUrl(JSONObject request) {
+        try {
+            if (request.getString(URL_PROPERTY) != null) {
+                return request.getString(URL_PROPERTY);
+            }
+        } catch (JSONException e) {
+            return "";
+        }
+        return "";
+    }
+
+    private String extractMethod(JSONObject request) {
+        try {
+            if (request.getString(METHOD_PROPERTY) != null) {
+                return request.getString(METHOD_PROPERTY);
+            }
+        } catch (JSONException e) {
+            return "";
+        }
+        return "";
+    }
+
+    private String extractPayload(JSONObject request) {
+        try {
+            if (request.getString(PAYLOAD_PROPERTY) != null) {
+                return request.getString(PAYLOAD_PROPERTY);
+            }
+        } catch (JSONException e) {
+            return "";
+        }
+        return "";
+    }
+
+    private static String getResponseAsString(HttpResponse response) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(response.getStatusLine().toString());
+        builder.append("\r\n");
+        for (Header header : response.getAllHeaders()) {
+            builder.append(header.getName());
+            builder.append("=");
+            builder.append(header.getValue());
+            builder.append("\r\n");
+        }
+        builder.append("\r\n");
+        if (response.getEntity() != null) {
+            try {
+                builder.append(EntityUtils.toString(response.getEntity()));
+            } catch (IOException ignore) {
+            }
+        }
+        return builder.toString();
+    }
+
+    private HashMap extractHeaders(JSONObject request) {
+        HashMap<String, String> headersMap = new HashMap();
+        try {
+            if (request.getJSONArray(HEADERS_PROPERTY) != null) {
+                JSONArray headersArray = request.getJSONArray(HEADERS_PROPERTY);
+                for (int i = 0; i < headersArray.length(); i++) {
+                    JSONArray headerGroup = (JSONArray) headersArray.get(i);
+                    String headerName = (String) headerGroup.get(0);
+                    if (StringUtils.hasContent(headerName)) {
+                        String headerValue = (String) headerGroup.get(1);
+                        headersMap.put(headerName, headerValue);
+                    }
+                }
+            }
+        } catch (JSONException e) {
+            return headersMap;
+        }
+        return headersMap;
+    }
+
+    private void setHeaders(AbstractHttpMessage message, HashMap<String, String> headersMap) {
```

Comment:
HashMap -> Map it doesn't make sense to show the details of implementation in the method definition